### DISCUSSION TITLE
How to get azure environment name when deploying?

### DISCUSSION THREAD

**Discussion by RamType0** [microsoft/aspire#5875] [Q&A]
How to get azure environment name when deploying?
I want to use this like
```C#
if (builder.ExecutionContext.IsPublishMode)
{
    var azureEnvironmentName = builder.Configuration["AZURE_ENV_NAME"] ?? throw new InvalidOperationException();
    var environmentName = builder.Configuration[$"Environments:{azureEnvironmentName}"] ?? throw new InvalidOperationException();
    var environmentVariableName = "DOTNET_ENVIRONMENT";
    apiService.WithEnvironment(environmentVariableName, environmentName);
    webFrontend.WithEnvironment(environmentVariableName, environmentName);
}
```

Currently, if I run this code, `builder.Configuration["AZURE_ENV_NAME"] ` returns null.

**Comment by davidfowl** [comment]
Ddi you read that AZURE_ENV_NAME is set somewhere? How did you come up with that?

**Reply by RamType0** [reply]
[In this article](https://learn.microsoft.com/ja-jp/dotnet/aspire/deployment/azure/aca-deployment-azd-in-depth?tabs=windows#initialize-the-template), it is said that
> azd generates a number of files and places them into the working directory. These files are:
> - azure.yaml: Describes the services of the app, such as .NET Aspire AppHost project, and maps them to Azure resources.
> - .azure/config.json: Configuration file that informs azd what the current active environment is.
> - .azure/aspireazddev/.env: Contains environment specific overrides.

and then 
`AZURE_ENV_NAME="MyApp-Staging"`
was in .env file.

**Reply by davidfowl** [reply]
cc @vhvb1989

**Reply by RamType0** [reply]
Note:I'm using VisualStudio for deploying.

**Reply by vhvb1989** [reply]
Try changing `AZURE_ENV_NAME`  to `DOTNET_ENVIRONMENT`

See:
![image](https://github.com/user-attachments/assets/43ccce08-720c-481d-b47d-74dc20238d4f)

https://github.com/Azure/azure-dev/blob/22a3496692fa61b81273260f75881f91046833e9/cli/azd/pkg/tools/dotnet/dotnet.go#L148

**Reply by vhvb1989** [reply]
Also found this:

![image](https://github.com/user-attachments/assets/64a66b25-1d56-4bf0-9a1e-d754049a4fac)

https://github.com/Azure/azure-dev/blob/22a3496692fa61b81273260f75881f91046833e9/cli/azd/internal/vsrpc/environment_service_create.go#L48

**Reply by RamType0** [reply]
DOTNET_ENVIRONMENT was "Development",
ASPIRE_ENVIRONMENT was null.
It is because my launchSettings.json is 
```JSON
{
  "$schema": "https://json.schemastore.org/launchsettings.json",
  "profiles": {
    "https": {
      "commandName": "Project",
      "dotnetRunMessages": true,
      "launchBrowser": true,
      "applicationUrl": "https://localhost:17229;http://localhost:15182",
      "environmentVariables": {
        "ASPNETCORE_ENVIRONMENT": "Development",
        "DOTNET_ENVIRONMENT": "Development",
        "DOTNET_DASHBOARD_OTLP_ENDPOINT_URL": "https://localhost:21149",
        "DOTNET_RESOURCE_SERVICE_ENDPOINT_URL": "https://localhost:22153"
      }
    },
    "http": {
      "commandName": "Project",
      "dotnetRunMessages": true,
      "launchBrowser": true,
      "applicationUrl": "http://localhost:15182",
      "environmentVariables": {
        "ASPNETCORE_ENVIRONMENT": "Development",
        "DOTNET_ENVIRONMENT": "Development",
        "DOTNET_DASHBOARD_OTLP_ENDPOINT_URL": "http://localhost:19187",
        "DOTNET_RESOURCE_SERVICE_ENDPOINT_URL": "http://localhost:20183"
      }
    }
  }
}

```

Visual Studio always uses first launch profile for Aspire deploying,
So these value has been provided from "https" profile.
